### DISCUSSION TITLE
Minimal API performance when parsing routes like "/search/{name}:{min:float}-{max:float}"

### DISCUSSION THREAD

**Discussion by audacode** [dotnet/aspnetcore#57155] [General]
I was just wondering if the ASP.NET team have been heroes yet and are parsing this type of route using Span(s), etc.

So for this example:
```cs
app.MapGet("/search/{name}:{min:float}-{max:float}", (string? name, float min, float max) => new { name, min, max });
```

Is the internal implementation using string splitting, etc.  Or does the internal code ultimately use Span<char>, with Range and the modern Split extension methods that work with spans rather than creating internal strings and arrays?

I was just looking at someone else's benchmark where they were comparing ASP.NET and showing it slower than various other frameworks, and looked at their code.  They were manually parsing the type of URI shown above, so I was just wondering if they used a URL like above if the internal ASP.NET implementation would do a fast modern parsing of that data into the floats, or if it would also do a bunch of allocations of strings and arrays while it does the parsing?

**Comment by gfoidl** [comment]
Put (very) simple at the moment it goes like this:
1. parsing of the request into `RouteValues`
2. parsing of `T` out of that `RouteValues` as `string`

> parsing this type of route using Span(s), etc.

For this you may be interested in https://github.com/dotnet/aspnetcore/issues/55016
